tests/helper.py

```diff
@@ -0,0 +1,33 @@
+import random
+import itertools
+import string
+
+
+def get_error_string(errors):
+    err_str = '\n'
+    for error in errors:
+        err_str += error + '\n\n'
+    return err_str
+
+
+def gen_random_alphabet_string(length=30):
+    s = ''
+    for i in range(0, length):
+        s += random.choice(string.ascii_letters)
+
+    return s
+
+
+def gen_all_possible_pair(iterable, beg=0, end=-1):
```

Comment:
Instead of `-1`, I would just use `None`, as that is more idiomatic. Especially since `-1` is a valid index in python. 😆